File: pkgs/primer_py/primer_py/ejemplo_1.py
#! /usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import Int32

class MyNodo(Node):
    def __init__(self):
        super().__init__('primer_node')
        self.count_ = 2
        self.publisher_  = self.create_publisher(Int32,'publisher_topic',10)
        self.timer_ = self.create_timer(1.0, self.callback_function)
        self.get_logger().info('Nodo publicador activo')


    def callback_function(self):
        message = Int32()
        message.data = self.count_
        self.publisher_.publish(message)

def main(args= None):
    rclpy.init(args=args)
    node = MyNodo()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: pkgs/primer_cpp/src/ejemplo_1.cpp

#include <chrono>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int32.hpp"

using namespace std::chrono_literals;

class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node("publicador_cpp"), count_(0)
    {
        publisher_ = this->create_publisher<std_msgs::msg::Int32>("publisher_topic", 10);
        timer_ = this->create_wall_timer(1s, std::bind(&MyNode::timer_callback, this));
    }

private:
    void timer_callback()
    {
        auto message = std_msgs::msg::Int32();
        message.data = 3;
        RCLCPP_INFO(this->get_logger(), "Publicando '%d'", message.data);
        publisher_->publish(message);

    }

    rclcpp::Publisher<std_msgs::msg::Int32>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    size_t count_;   
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MyNode>());
    rclcpp::shutdown();
    return 0;
}


